common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -803,21 +828,25 @@ public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
                 break;
             case '.':
                 ++ipv4Seperators;
-                if (i - begin > IPV4_MAX_CHAR_BETWEEN_SEPARATOR
+                tmp = i - begin; // tmp is the length of the current segment.
+                if (tmp > IPV4_MAX_CHAR_BETWEEN_SEPARATOR
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (19) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)